### PR TITLE
Xeno devour refactor

### PR DESCRIPTION
**Pull request by Rohesie**
Closes #1877
Same solution but using a signal.

### CODE REVIEW

**Comment by DominikPanic** [comment]
Feel free to add Closes: #1877 in the description

**Comment by Rohesie** [comment]
Done.

File: code/modules/mob/living/carbon/xenomorph/death.dm [hunk 1]
```diff
@@ -32,6 +32,7 @@
 	for(var/atom/movable/A in stomach_contents)
 		stomach_contents.Remove(A)
 		A.forceMove(loc)
+		SEND_SIGNAL(A, COMSIG_LIVING_RELEASED_FROM_STOMACH, src)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/death.dm]: A signal must have only one source, make a different signal type

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -89,6 +89,7 @@
 	attack_icon = image("icon" = 'icons/effects/attacks.dmi',"icon_state" = "", "layer" = 0)
 	GLOB.mob_living_list += src
 	START_PROCESSING(SSmobs, src)
+	RegisterSignal(src, COMSIG_LIVING_RELEASED_FROM_STOMACH, .proc/on_release_from_stomach)
```
Comment (spookydonut) [inline code/modules/mob/living/living.dm]: Should be registered when the xeno swallows them and removed when released

**Pull request merged**